### PR TITLE
[ENH] Implemented logic for node selection

### LINKED ISSUE #3: Implement Node selection
- [x] fetch the list of nodes from the API (with error handling)
- [x] default selection is all
- [x] User can select nodes and that becomes part of the request

### PR DESCRIPTION
**Pull request by rmanaem**
- Closes #3 

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -325,17 +413,36 @@ function App() {
   }
 
   return (
-    <div className="grid grid-cols-4 grid-rows-1 gap-4">
+    <div className='grid grid-cols-4 grid-rows-1 gap-4'>
```
Comment (surchs) [inline src/App.tsx:416]: I guess this is prettier's doing? We should probably add that to the husky runner. Could you make an issue?

**Review by surchs — APPROVED** [review]
very nice, looks good to me!

:cook:

File: src/App.tsx (hunk 1 above)
Comment (rmanaem) [reply src/App.tsx:416]: No that was actually me but I will make an issue for the prettier.

**Comment by surchs** [comment]
:rocket: PR was released in [`v0.1.0`](https://github.com/neurobagel/query-tool/releases/tag/v0.1.0) :rocket:

**Pull request merged**